homeassistant/helpers/entity.py

```diff
@@ -473,8 +473,9 @@ def async_on_remove(self, func: CALLBACK_TYPE) -> None:
             self._on_remove = []
         self._on_remove.append(func)
 
-    async def async_remove(self):
+    async def async_remove(self) -> None:
         """Remove entity from Home Assistant."""
+        assert self.hass is not None
```

Comment:
We should avoid using `assert` in our code base, since it is not run when optimization flags are used on Python.

Reply:
It's really only to validate mypy